src/niscope/system_tests/test_system_niscope.py

```diff
@@ -66,6 +67,23 @@ def test_fetch_binary8_into(session):
     assert len(wfm_infos) == test_num_channels
 
 
+def test_fetch_binary8(session):
+    test_voltage = 1.0
+    test_record_length = 2000
+    test_channels = range(2)
+    test_num_channels = 2
+    wfm = numpy.ndarray(test_num_channels * test_record_length, dtype=numpy.int8)
+    # Initialize with NaN so we can later verify all samples were overwritten by the driver.
+    wfm.fill(float('nan'))
+    session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
+    session.configure_horizontal_timing(50000000, test_record_length, 50.0, 1, True)
+    with session.initiate():
+        wfm_infos = session.channels[test_channels].fetch(wfm=wfm)
```

Comment:
Interesting.
You're using type of wfm to determine which binary flavor to call into? I was under the impression fancy fetch was ease of use so it should only support float. In the name of KISS I would remove this flexibility. If you think we should keep it, I'd like to use a different way of selecting which underlying thing to call because it's odd to modify the input param _and_ also return it (or memory views to it?).

Client wants fast? Call the appropriate fetch_into. Or use C, really.

Reply:
If you have to call _into, you don't get the ease of use benefits. 

We don't really return the same parameter that was passed in. We return either new `array.array`'s or `memoryview`'s of the array passed in, but these correspond to the acquired records and are fairly transformed from what was originally passed in.

It seems like the benefits of the fancy fetch should be available along with the speed of numpy and _into.

Reply:
"fancy fetch" is also "easy to use fetch"

I expect +90% of users to want scaled float results. They will use "fancy fetch".
I expect only a very small share of users to want to extract absolute maximum performance and don't need things scaled for some rare reason. They will use something other than float and the "fetch_into" methods.
Users who want "easy to use fetch" _and_ unscaled data will be extremely rare if any. I don't think adding complexity to "fancy fetch"'s usage that impact +90% of the users is the right tradeoff.

Reply:
I don't think it add much complexity at all. Not at the API level (one additional, optional parameter) or in the implementation. 
I do think the additional benefits of encapsulating the property calls as well as bundling the actual waveform with the rest of the waveform information for each individual record is worth it.